cmd/yggd/main.go

```diff
@@ -26,6 +27,371 @@ var (
+		log.Errorf("cannot send connection status message: %v", err)
+	}
+}
+
+// monitorCanonicalFacts tries to monitor canonical facts file for changes
+func monitorCanonicalFacts(client *Client) {
+	if config.DefaultConfig.CanonicalFacts == "" {
+		return
+	}
+	c := make(chan notify.EventInfo, 1)
+	if err := notify.Watch(config.DefaultConfig.CanonicalFacts, c, notify.InCloseWrite); err != nil {
+		log.Infof("cannot start watching '%v': %v", config.DefaultConfig.CanonicalFacts, err)
+		return
+	}
+	defer notify.Stop(c)
+
+	for e := range c {
+		switch e.Event() {
+		case notify.InCloseWrite:
+			go func() {
+				msg, err := client.ConnectionStatus()
+				if err != nil {
+					log.Fatalf("cannot get connection status: %v", err)
+				}
+				if _, _, _, err := client.SendConnectionStatusMessage(msg); err != nil {
+					log.Errorf("cannot send connection status message: %v", err)
+				}
+			}()
+		}
+	}
+}
+
+// monitorTags tries to monitor tags file for changes
+func monitorTags(client *Client) {
+	c := make(chan notify.EventInfo, 1)
+
+	fp := filepath.Join(constants.ConfigDir, "tags.toml")
+
+	if err := notify.Watch(fp, c, notify.InCloseWrite, notify.InDelete); err != nil {
+		log.Infof("cannot start watching '%v': %v", fp, err)
+		return
+	}
+	defer notify.Stop(c)
+
+	for e := range c {
+		log.Debugf("received inotify event %v", e.Event())
+		switch e.Event() {
+		case notify.InCloseWrite, notify.InDelete:
+			go func() {
+				msg, err := client.ConnectionStatus()
+				if err != nil {
+					log.Fatalf("cannot get connection status: %v", err)
+				}
+				if _, _, _, err = client.SendConnectionStatusMessage(msg); err != nil {
+					log.Errorf("cannot send connection status: %v", err)
+				}
+			}()
+		}
+	}
+}
+
+// monitorCertificate tries to monitor certificate file for changes
+func monitorCertificate(
+	TlSEvents chan *tls.Config,
+	transporter transport.Transporter,
+	dispatcher *work.Dispatcher,
+) {
+	// Can be that there are no files to watch
+	if TlSEvents == nil {
+		log.Info("no TLS configuration, disabling TLS watcher update")
+		return
+	}
+
+	for cfg := range TlSEvents {
+		log.Debug("reloading transport TLS configuration")
+		err := transporter.ReloadTLSConfig(cfg)
+		if err != nil {
+			log.Errorf("cannot update transporter TLS configuration: %v", err)
+			continue
+		}
+		log.Info("transport TLS configuration reloaded")
+
+		log.Debug("setting dispatcher HTTP client")
+		httpClient := http.NewHTTPClient(cfg, UserAgent)
+		dispatcher.HTTPClient = httpClient
+		log.Info("dispatcher HTTP client updated")
+	}
+}
+
+// systemdWatchDog tries to send sd_notify to systemd.
+// More details about sd_notify can be found here:
+// https://www.freedesktop.org/software/systemd/man/sd_notify.html
+func systemdWatchDog() {
+	watchdogDuration, err := daemon.SdWatchdogEnabled(false)
+	if err != nil {
+		log.Errorf("cannot get watchdog duration: %v", err)
+	}
+	if watchdogDuration > 0 {
```

Comment:
I feel like I want to see a debug message here.

```suggestion
	if watchdogDuration > 0 {
        log.Debug("starting systemd watchdog notification")
```

Reply:
Debug message added.